src/mixins/filing-template-mixin.ts

```diff
@@ -391,33 +398,47 @@ export default class FilingTemplateMixin extends DateMixin {
         foundingDate: this.getEntitySnapshot.businessInfo.foundingDate,
         identifier: this.getEntitySnapshot.businessInfo.identifier,
         legalName: this.getEntitySnapshot.businessInfo.legalName,
-        legalType: this.getEntitySnapshot.businessInfo.legalType
+        legalType: this.getEntitySnapshot.businessInfo.legalType,
+        nrNumber: this.getEntitySnapshot.businessInfo.nrNumber
       },
       specialResolution: {
         ...this.getSpecialResolution
       }
     }
 
-    /* Only add alteration if the association type has changed,
-     * rules and memorandum only show up if the association type changes. */
-    if (this.hasAssociationTypeChanged) {
+    // Only add alteration if the association type or rules or memorandum have changed
+    if (this.hasAssociationTypeChanged || this.hasMemorandumChanged || this.hasRulesChanged) {
+      // Conditional access - Rules or memorandum could be empty here if it was done on paper.
       filing.alteration = {
         business: {
           identifier: this.getBusinessId,
           legalType: this.getEntityType
         },
         contactPoint: this.getContactPoint,
         cooperativeAssociationType: this.getAssociationType,
-        rulesFileKey: 'test',
-        rulesFileName: 'testUrl',
-        memorandumFileKey: 'test',
-        memorandumFileName: 'test'
+        rulesFileKey: this.getRules?.key,
+        rulesFileName: this.getRules?.name,
+        memorandumFileKey: this.getMemorandum?.key,
+        memorandumFileName: this.getMemorandum?.name
+      }
+      if (this.getMemorandum?.includedInResolution) {
```

Comment:
Comment to explain what you're doing here (or why).

Also line 429.

Reply:
Yup, plus I'll link to the LEAR validator code that is relevant to this

Reply:
Thanks for the comment.